src/controllers/portfolio/portfolio.ts

```diff
@@ -72,10 +63,25 @@ function getHintsWithBalance(
     ])
   )
 
-  return {
-    erc20s,
-    erc721s
+  storagePreviousHints.fromExternalAPI = {
+    ...storagePreviousHints.fromExternalAPI,
+    [key]: { erc20s, erc721s }
   }
+
+  // Set lastSeenNonZero timestamp for learnedTokens
+  erc20s.forEach((address) => {
+    storagePreviousHints.learnedTokens = {
+      ...(storagePreviousHints.learnedTokens || {}),
+      [networkId]: {
+        ...((storagePreviousHints?.learnedTokens &&
+          storagePreviousHints?.learnedTokens[networkId]) ??
+          {}),
+        [address]: Date.now().toString()
+      }
+    }
+  })
```

Comment:
Pretty sure this code can be written in a more readable and performant way. Example(haven't tested it):
``` 
if (!storagePreviousHints.learnedTokens[networkId]) {
  storagePreviousHints.learnedTokens[networkId] = {}
}
erc20s.forEach((address) => {
  storagePreviousHints.learnedTokens[networkId][address] = Date.now().toString()
})
```